src/ptr_operator.rs

```diff
@@ -19,7 +17,7 @@ pub fn ptr_operands_read(
     }
 
     let pointer = FatPointer::decode(src0.value);
-    if src1.value > MAX_OFFSET_FOR_ADD_SUB {
+    if src1.value > u32::MAX.into() {
```

Comment:
`MAX_OFFSET_FOR_ADD_SUB == u32::MAX` 

Am I missing something?

Reply:
Yes I know, but for some reason that was not working

Reply:
`MAX_OFFSET_FOR_ADD_SUB` is `u32:MAX + 1`

Reply:
@gianbelinche is correct